util/crashReport.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# TODO
+# remove 'logging' echo statements
+
+# Variables for script usage
+FILE_HS_ERR_LOG=/opt/instana/agent/hs_err.log
+DIR_AGENT_LOG=/opt/instana/agent/data/log
+AGENT_LOG_COLLECT_LINES=1000
+
+# Variables for use in the JSON payload
+JSON_HOSTNAME="$(hostname)"
+JSON_TIMESTAMP="$(date +%s)000"
+JSON_PID="${1}"
+if [ "${2}" != "" ]; then
+  JSON_ERROR="${2}"
+else
+  JSON_ERROR="Unknown reason for Agent crash"
+fi
+
+
+post_crash_data() {
+  echo
+  echo $(generate_crash_data)
+  echo
+  # Silence all output so it doesn't show up in e.g. Docker or Kubernetes logs
+  curl -XPOST "https://${INSTANA_AGENT_ENDPOINT}:${INSTANA_AGENT_ENDPOINT_PORT}/metrics" \
+    --silent \
+	--http2-prior-knowledge \
+	--header "x-instana-key: ${INSTANA_AGENT_KEY}" \
+	--header "x-instana-host: ${JSON_HOSTNAME}" \
+	--header "Content-Type: application/json" \
+	--data "$(generate_crash_data)" \
+    --output /dev/null 2>&1
+  }
+
```

Comment:
You might consider adding --max-time, so the agent does spend any more time than it needs to before exiting in case of a network event.